src/hooks/react/useDebounceState.ts

```diff
@@ -0,0 +1,52 @@
+// Copyright 2021 Parity Technologies (UK) Ltd.
+// This file is part of Parity Bridges UI.
+//
+// Parity Bridges UI is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Parity Bridges UI is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Parity Bridges UI.  If not, see <http://www.gnu.org/licenses/>.
+
+import { useCallback, useState } from 'react';
+import debounce from 'lodash/debounce';
+
+type Output = [string | null, string | null, Function];
+
+export const useDebounceState = (
+  initialValue: string | null,
+  wait?: null | number,
+  transformCallback?: Function
+): Output => {
+  const [value, setValue] = useState(initialValue);
+  const [debounced, setDebounced] = useState(value);
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Looks wrong.

Reply:
It's not i'm afraid ... if i don't disable the linter on that i receive the following message:

```
React Hook useCallback received a function whose dependencies are unknown. Pass an inline function instead
```

But the problem is that i need the debounce enclosing the whole function otherwise it does not work. Any ideas how to solve this without disabling eslint?

Reply:
Well, `useCallback` and `debounce` sounds like a common problem, I found some solution [here](https://kyleshevlin.com/debounce-and-throttle-callbacks-with-react-hooks) do you think that could work?

Reply:
I wanted to say `use useMemo` but link of Tomasz gave that away - and it would sound like I copied it from there - so I say nothing.

Reply:
works like charm ... thanks both!